enhancements/insights/rapid-recommendations.md

```diff
@@ -0,0 +1,549 @@
+---
+title: rapid-recommendations
+authors:
+  - "@jholecek-rh"
+  - "@tremes"
+reviewers: 
+  - "@deads2k"
+approvers: 
+  - "@deads2k"
+api-approvers: 
+  - None
+creation-date: 2024-01-03
+last-updated: 2024-02-22
+tracking-link: 
+  - https://issues.redhat.com/browse/CCXDEV-12213
+  - https://issues.redhat.com/browse/CCXDEV-12285
+see-also:
+  - "/enhancements/insights/conditional-data-gathering.md"
+replaces:
+  - None
+superseded-by:
+  - None
+---
+
+# Insights Rapid Recommendations
+
+This enhancement proposal introduces a new approach to how the data collected 
+by the Insights operator can be defined remotely.
+
+
+## Summary
+
+The Insights Operator collects various data and resources from the OpenShift and Kubernetes APIs. 
+The definition of the collected data is mostly hardcoded in the operator's source code and largely 
+locked in the corresponding OCP version.
+
+This proposal introduces remote Insights Operator configuration for collected data. 
+The feature will allow Red Hat to control what data the Insights Operator gathers, 
+within hardcoded boundaries, independently of the cluster version. 
+
+
+## Motivation
+
+The Insights Operator collects various data and resources from the OpenShift and Kubernetes APIs. 
+The definition of the collected data is mostly hardcoded in the operator's source code 
+and largely locked in the corresponding OCP version.
+
+The data gathered by the Insights Operator can be divided into several groups based on their source 
+and format:
+
+* API resources - API resources are identified by the group, version, kind (GVK) and, optionally, 
+  namespace and name. 
+  They are represented as JSON or YAML objects.
+* Container logs - Container logs are identified by the namespace name, pod name, and messages of 
+  interest to be filtered from the log. 
+  They are represented as line-oriented text files.
+* Node logs - Similar to container logs, but identified/queried using different parameters.
+* Prometheus metrics and alerts - See the 
+  [MostRecentMetrics](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#mostrecentmetrics) 
+  and [ActiveAlerts](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#activealerts) gatherers.
+* Aggregated data - This is data where the Insights Operator employs data processing logic 
+  that produces a custom representation of raw data. 
+  See e.g. the [WorkloadInfo](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#workloadinfo)
+  and [HelmInfo](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#helminfo) gatherers.
+
+When an application (e.g. an Insights recommendation) requests new data, a new Insights Operator 
+component (a gatherer) must be developed and, optionally, backported to supported y-stream OCP versions.
+This is a labor-intesive and error-prone process. 
```

Comment:
For the record (we discussed this offline), the main challenges in this respect are:

* Maintaining a consistent and predictable archive structure. The current structure is not predictable. It means that data consumers have to manually map the archive content back to cluster data it represents. This is most pronounced for (custom) resources.
* Maintaining archive structure semantics. For data consumers to be efficient, they need to rely on paths to know where to find data they need. We had a few issues in the past where new gatherers caused some paths to become ambiguous.
* Maintaining information about data availability. Because of the development process, information about data availability for individual data points is maintained manually and added retrospectively. We've had several issues where the information was inaccurate, leading to incorrect decisions by or delays for data consumers.